### PR TITLE
[IOTDB-2295][IOTDB-2299] Fix Timeseries count group by level Bug & Fix Wrong SQL instance

### PR DESCRIPTION
**Pull request by zyk990424**
### Problem
Since introducing PathPattern-Match feature, the original implementation of "count timeseries group by level" won't work any more since it relies on prefix-match

### Solution
Implement "count timeseries group by level" in MTree via MeasurementGroupByLevelCounter.
During MTree traversal, the node matching the given level will be stored and the measurement matching the given pattern will be counted in it's belonged node.

### Others
This pr also fix some wrong sql instances in UserGuide docs.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/mtree/traverser/counter/MeasurementGroupByLevelCounter.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    this.groupByLevel = groupByLevel;
+  }
+
+  @Override
+  protected boolean processInternalMatchedMNode(IMNode node, int idx, int level)
+      throws MetadataException {
+    if (level == groupByLevel) {
+      path = node.getPartialPath();
+      if (!result.containsKey(path)) {
+        result.put(path, 0);
```
Comment (SilverNarcissus) [inline server/src/main/java/org/apache/iotdb/db/metadata/mtree/traverser/counter/MeasurementGroupByLevelCounter.java:51]: You can use putIfAbsent method to save look up time

File: server/src/main/java/org/apache/iotdb/db/metadata/mtree/traverser/counter/MeasurementGroupByLevelCounter.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    return false;
+  }
+
+  @Override
+  protected boolean processFullMatchedMNode(IMNode node, int idx, int level)
+      throws MetadataException {
+    if (level == groupByLevel) {
+      path = node.getPartialPath();
+      if (!result.containsKey(path)) {
+        result.put(path, 0);
```
Comment (SilverNarcissus) [inline server/src/main/java/org/apache/iotdb/db/metadata/mtree/traverser/counter/MeasurementGroupByLevelCounter.java:63]: You can use putIfAbsent method to save look up time

**Review by SilverNarcissus — APPROVED** [review]
Brilliant work!  Only one issue, please have a look.

File: server/src/main/java/org/apache/iotdb/db/metadata/mtree/traverser/counter/MeasurementGroupByLevelCounter.java (hunk 1 above)
Comment (zyk990424) [reply server/src/main/java/org/apache/iotdb/db/metadata/mtree/traverser/counter/MeasurementGroupByLevelCounter.java]: Fixed

File: server/src/main/java/org/apache/iotdb/db/metadata/mtree/traverser/counter/MeasurementGroupByLevelCounter.java (hunk 2 above)
Comment (zyk990424) [reply server/src/main/java/org/apache/iotdb/db/metadata/mtree/traverser/counter/MeasurementGroupByLevelCounter.java]: Fixed

**Review by JackieTien97 — CHANGES_REQUESTED** [review]
Better to add an IT.

**Comment by zyk990424** [comment]
> Better to add an IT.

IT added.

**Pull request merged**